flink-connector/src/main/java/com/datastrato/gravitino/flink/connector/catalog/GravitinoCatalogManager.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.flink.connector.catalog;
+
+import com.datastrato.gravitino.Catalog;
+import com.datastrato.gravitino.NameIdentifier;
+import com.datastrato.gravitino.client.GravitinoAdminClient;
+import com.datastrato.gravitino.client.GravitinoMetalake;
+import com.google.common.base.Preconditions;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** GravitinoCatalogManager is used to retrieve catalogs from Gravitino server. */
+public class GravitinoCatalogManager {
+  private static final Logger LOG = LoggerFactory.getLogger(GravitinoCatalogManager.class);
+  private static GravitinoCatalogManager gravitinoCatalogManager;
+
+  private volatile boolean isClosed = false;
+  private final String metalakeName;
+  private final GravitinoMetalake metalake;
```

Comment:
@FANNG1 I removed the cache. It indeed causes a consistency problem when other programs drop the gravitino catalog.